packages/lit-html/src/lib/parts.ts

```diff
@@ -0,0 +1,137 @@
+import {NodePart, nothing} from './lit-html.js';
+
+/**
+ * The state of a NodePart, which can be detached and reattached.
+ */
+export type NodePartState = {};
+
+/**
+ * The private interface for NodePartState, which should be kept opaque.
+ */
+type NodePartStateInternal = {
+  __value: unknown;
+  __fragment: DocumentFragment;
+};
+
+export const detachNodePart = (part: NodePart): NodePartState => {
+  const fragment = document.createDocumentFragment();
+  const state: NodePartStateInternal = {
+    __value: part.__value,
```

Comment:
Private only applies to class fields, but it's ok because we never export this interface. We export the `NodePartState` interface which doesn't have this property.